test/e2e/validation_test.go

```diff
@@ -475,22 +475,24 @@ func testServicesConfigMap(t *testing.T) {
 	operatorVersion, err := getWMCOVersion()
 	require.NoError(t, err)
 	servicesConfigMapName := servicescm.NamePrefix + operatorVersion
-	expected := &servicescm.Data{}
 
 	// Ensure the windows-services ConfigMap exists in the cluster
+	var expectedData *servicescm.Data
 	t.Run("Services ConfigMap existence", func(t *testing.T) {
-		_, err = tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Get(context.TODO(), servicesConfigMapName,
+		cm, err := tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Get(context.TODO(), servicesConfigMapName,
 			meta.GetOptions{})
-		assert.NoErrorf(t, err, "error ensuring ConfigMap %s exists", servicesConfigMapName)
+		require.NoErrorf(t, err, "error ensuring ConfigMap %s exists", servicesConfigMapName)
+		expectedData, err = servicescm.Parse(cm.Data)
```

Comment:
So you are assuming that at this point the data in the ConfigMap is what is expected and correct? Ideally you must be generating the expected data here based on some WMCO constant rather than reusing the `Parse` method. What happens if there is a bug in the `Parse` method?

Reply:
The difficulty in validating the correct data is why I made this change. The ConfigMap contents needs to be generated using the cluster configuration. To validate the contents we'd need to either duplicate the logic of, or export the `generateServiceManifest` function from the `controllers` package. I'd rather not do either of those as that leads to the issue you present of there possibly being a bug in the `generateServiceManifest` function, and us reusing that function isn't doing anything valuable by confirming that `generateServiceManifest` generates the same content when invoked in two different places.

I don't see an issue with using `Parse` as that is how the consumer of the ConfigMap, WICD, is going to interact with it.

In order to get some amount of validation here I've expanded the test to check that the required services exist within the ConfigMap.